src/components/List/List.js

```diff
@@ -1,21 +1,94 @@
 });
 
+const isListItem = element => element && element.type !== ListItem && element.type !== 'li';
+const NAVIGATION_STEP_VALUES = {
+    [LIST_NAVIGATION_DIRECTIONS.UP]: -1,
+    [LIST_NAVIGATION_DIRECTIONS.DOWN]: 1
+};
+
 const List = React.forwardRef((props, ref) => {
-    const { children, isDivided, ...rest } = props;
+    const { children, isDivided, onNavigate, onSelect, ...rest } = props;
+    const [selectedIndex, setSelectedIndex] = useState(null);
+
+    const getNextSelectedIndex = keyCode => {
+        const stepValue = NAVIGATION_STEP_VALUES[keyCode];
+        const nextSelectedIndex = selectedIndex + stepValue;
+
+        // Return 0 index if nextSelectedIndex has negative value or selectedIndex hasn't been updated before
+        if (nextSelectedIndex < 0 || selectedIndex === null) {
+            return 0;
+        }
+
+        // Return last React.Children index if nextSelectedIndex is out of the right bound
+        if (nextSelectedIndex >= children.length) {
+            return children.length - 1;
+        }
+
+        // Return nextSelectedIndex without any changes for others cases
+        return nextSelectedIndex;
+    };
+
+    const handleKeyDown = e => {
+        // Update selectedIndex with arrow navigation and make onNavigate function callback
+        if (e.key === LIST_NAVIGATION_DIRECTIONS.UP || e.key === LIST_NAVIGATION_DIRECTIONS.DOWN) {
+            const nextSelectedIndex = getNextSelectedIndex(e.key);
+
+            if (selectedIndex !== nextSelectedIndex) {
+                e.preventDefault();
+                setSelectedIndex(getNextSelectedIndex(e.key));
+
+                if (onNavigate) {
+                    onNavigate(e.key);
+                }
+            }
+        }
+
+        // Imitate onClick event on Enter press and make onSelect function callback
+        if (e.key === ENTER_KEY) {
+            if (
+                children[selectedIndex] &&
+                children[selectedIndex].props &&
+                children[selectedIndex].props.onClick
+            ) {
+                children[selectedIndex].props.onClick(e);
+
+                if (onSelect) {
+                    onSelect();
+                }
+            }
+        }
+    };
+
+    const handleMouseOver = index => {
+        if (selectedIndex !== index) {
+            setSelectedIndex(index);
+        }
+    };
+
     return (
-        <ul {...rest} ref={ref} {...block(props)}>
-            {React.Children.map(children, child =>
-                child ? React.cloneElement(child, elem('item', props)) : null
+        <ul {...rest} ref={ref} tabIndex="0" onKeyDown={handleKeyDown} {...block(props)}>
+            {React.Children.map(children, (child, index) =>
+                child
+                    ? React.cloneElement(child, {
+                          ...elem('item', props),
+                          isHighlighted: index === selectedIndex,
+                          onMouseOver: () => handleMouseOver(index)
```

Comment:
maybe enough to select an item on mouse enter? that would reduce confusion when the mouse is just forgotten over an item